website/templates/project/view_file.mako

```diff
@@ -1,5 +1,17 @@
 <%inherit file="project/project_base.mako"/>
 <%def name="title()">${file_name | h}</%def>
+
+% if (user['can_comment'] or node['has_comments']):
+    % if provider in ['github', 'figshare', 'dropbox', 's3', 'osfstorage']:
```

Comment:
This should probably be handled by a config variable in each addon's `__init__.py` file (e.g. `ALLOW_COMMENTS` rather than hardcoding them here.